sweepbatcher/sweep_batcher_test.go

```diff
@@ -1176,6 +1176,156 @@ func testDelays(t *testing.T, store testStore, batcherStore testBatcherStore) {
 	checkBatcherError(t, runErr)
 }
 
+// testMaxSweepsPerBatch tests the limit on max number of sweeps per batch.
+func testMaxSweepsPerBatch(t *testing.T, store testStore,
+	batcherStore testBatcherStore) {
+
+	defer test.Guard(t, test.WithGuardTimeout(5*time.Minute))()
+
+	lnd := test.NewMockLnd()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sweepStore, err := NewSweepFetcherFromSwapStore(store, lnd.ChainParams)
+	require.NoError(t, err)
+
+	startTime := time.Date(2018, 11, 1, 0, 0, 0, 0, time.UTC)
+	testClock := clock.NewTestClock(startTime)
+
+	// Create muSig2SignSweep failing all sweeps to force non-cooperative
+	// scenario (it increases transaction size).
+	muSig2SignSweep := func(ctx context.Context,
+		protocolVersion loopdb.ProtocolVersion, swapHash lntypes.Hash,
+		paymentAddr [32]byte, nonce []byte, sweepTxPsbt []byte,
+		prevoutMap map[wire.OutPoint]*wire.TxOut) (
+		[]byte, []byte, error) {
+
+		return nil, nil, fmt.Errorf("test error")
+	}
+
+	// Set publish delay.
+	const publishDelay = 3 * time.Second
+
+	batcher := NewBatcher(
+		lnd.WalletKit, lnd.ChainNotifier, lnd.Signer,
+		muSig2SignSweep, testVerifySchnorrSig, lnd.ChainParams,
+		batcherStore, sweepStore, WithPublishDelay(publishDelay),
+		WithClock(testClock),
+	)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var runErr error
+	go func() {
+		defer wg.Done()
+		runErr = batcher.Run(ctx)
+	}()
+
+	// Wait for the batcher to be initialized.
+	<-batcher.initDone
+
+	const swapsNum = 2500
```

Comment:
To reduce test runtime, would just 1001 work? Also the test seems to be logging excessively, so perhaps we could reduce that too. wdyt?

Reply:
Great idea! I changed the number of swaps to 1001. Disabling logging has negligible effect.

Reply:
Disabling logging is helpful when running `make unit` locally, as it's easier to follow test logs.

Reply:
Disabled logging in the test.